cra-rxjs-styled-components/src/hooks/gists/use-gists.ts

```diff
@@ -0,0 +1,87 @@
+import { useEffect, useState } from 'react';
+import { GITHUB_URL_BASE, GISTS_URL } from '../../constants/url.constants';
+import { Gist, GistWithFilename } from '../../interfaces/gists.interfaces';
+import { fromFetchWithAuth } from '../auth/from-fetch-with-auth';
+import {
+  filter,
+  map,
+  Subscription,
+  Observable,
+  switchMap,
+  tap,
+  zip,
+} from 'rxjs';
+
+export function useGists(): GistWithFilename[] {
+  const [state, setState] = useState<GistWithFilename[]>([]);
+  const [currentUser, setCurrentUser] = useState<string>('');
+
+  const getCurrentUser = () =>
+    fromFetchWithAuth<{
+      login: string;
+    }>(`${GITHUB_URL_BASE}/user`, {
+      selector: (response: Response) => {
+        return response.json();
+      },
+    });
+
+  useEffect(() => {
+    getCurrentUser().subscribe((user) => {
+      setCurrentUser(user.login);
+    });
```

Comment:
Is there not some global state for the logged in user available and just pass it in as a prop? I don't think we want to refetch the user in a bunch of different places

Reply:
There isn't but I can look into making one. This would just move this function elsewhere.

Reply:
@ktrz was talking to Ben Lesh about this. I don't know if they came up with a solution though.